### PR TITLE
[android] Enable nonTransitiveRClass feature

### PR DESCRIPTION
**Pull request by Jean-BaptisteC**
From Gradle 8.0, `android.nonTransitiveRClass` property is `true` by default in `gradle.properties`
Enable this feature, optimize R class, and reduce build time (I didn't see a difference).
To update all Android code, I have used `Migrate to Non-Transitive R Classes` feature in Android Studio.
https://developer.android.com/build/optimize-your-build#use-non-transitive-r-classes
https://medium.com/androiddevelopers/5-ways-to-prepare-your-app-build-for-android-studio-flamingo-release-da34616bb946

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -6,7 +6,7 @@ org.gradle.caching=true
 org.gradle.jvmargs=-Xmx1024m -Xms256m
 android.useAndroidX=true
 android.native.buildOutput=verbose
-android.nonTransitiveRClass=false
+android.nonTransitiveRClass=true
```
Comment (meenbeese) [inline android/gradle.properties:9]: > The nonTransitiveRClass flag in the gradle.properties file controls the R class behavior. From AGP 8.0, it’s true when not specified, and thus becomes the default.

We don't have to even specify it, right? It's already `true` by default.

Comment (biodranik) [reply android/gradle.properties:9]: Let's leave it for clarity.

**Review by biodranik — APPROVED** [review]
Did you test it on Android 5?

**Comment by Jean-BaptisteC** [comment]
Tested on Lenovo Yoga Tab 2 - Android 5.0

**Pull request merged**